R/jhu_data.R

```diff
@@ -68,102 +73,176 @@ load_jhu_data <- function(
-      'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine',
-      'Maryland', 'Massachusetts', 'Michigan', 'Minnesota',
-      'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada',
-      'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
-      'North Carolina', 'North Dakota', 'Northern Mariana Islands',
-      'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Puerto Rico',
-      'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee',
-      'Texas', 'Utah', 'Vermont', 'Virgin Islands', 'Virginia',
-      'Washington', 'West Virginia', 'Wisconsin', 'Wyoming')
+      "Alabama", "Alaska", "American Samoa", "Arizona", "Arkansas", "California",
+      "Colorado", "Connecticut", "Delaware", "District of Columbia",
+      "Florida", "Georgia", "Guam", "Hawaii", "Idaho", "Illinois",
+      "Indiana", "Iowa", "Kansas", "Kentucky", "Louisiana", "Maine",
+      "Maryland", "Massachusetts", "Michigan", "Minnesota",
+      "Mississippi", "Missouri", "Montana", "Nebraska", "Nevada",
+      "New Hampshire", "New Jersey", "New Mexico", "New York",
+      "North Carolina", "North Dakota", "Northern Mariana Islands",
+      "Ohio", "Oklahoma", "Oregon", "Pennsylvania", "Puerto Rico",
+      "Rhode Island", "South Carolina", "South Dakota", "Tennessee",
+      "Texas", "Utah", "Vermont", "Virgin Islands", "Virginia",
+      "Washington", "West Virginia", "Wisconsin", "Wyoming"
+    )
 
     state_results <- jhu_data %>%
       dplyr::filter(Province_State %in% states_to_keep) %>%
       dplyr::mutate(location_name = Province_State) %>%
       dplyr::group_by(location_name, date) %>%
       dplyr::summarize(cum = sum(cum)) %>%
       dplyr::group_by(location_name) %>%
-      dplyr::mutate(inc = cum - dplyr::lag(cum, 1L)) %>%
+      dplyr::mutate(inc = diff(c(0, cum))) %>%
       dplyr::ungroup() %>%
       dplyr::left_join(
         covidData::fips_codes %>% dplyr::filter(nchar(location) == 2),
-        by = 'location_name') %>%
+        by = "location_name"
+      ) %>%
       dplyr::select(location, date, cum, inc)
-    
+
     results <- dplyr::bind_rows(results, state_results)
   }
 
   # summarized results for national level
-  if('national' %in% spatial_resolution) {
-    # because we don't filter on states_to_keep as above, we are off by a total
+  if ("national" %in% spatial_resolution) {
+    # because we don"t filter on states_to_keep as above, we are off by a total
     # of 3 deaths attributed to Diamond Princess.
     national_results <- jhu_data %>%
       dplyr::group_by(date) %>%
       dplyr::summarize(cum = sum(cum)) %>%
       dplyr::ungroup() %>%
       dplyr::mutate(
-        inc = cum - dplyr::lag(cum, 1L),
-        location = 'US'
+        inc = diff(c(0, cum)),
+        location = "US"
       ) %>%
       dplyr::select(location, date, cum, inc)
 
     results <- dplyr::bind_rows(results, national_results)
   }
+
   
+   if (replace_negatives == TRUE) {
```

Comment:
This is fine (no need to change), but note you could also just use

```suggestion
   if (replace_negatives) {
```